### DISCUSSION TITLE
Generating SVG rather than data: ?

### DISCUSSION THREAD

**Discussion by alanef** [chillerlan/php-qrcode#208] [Q&A]
For reasons best known to me  :) ) I'm trying to avoid data: as in <img src="data:.... >

And I also dont want to store in  files cached or perm.

So I thought I want an SVG 

`$options->imageBase64 = false;`

Would be a starting point but it appears to generate png/base64

What am I missing here?

`  
                $qrcode =new QRCode;

		$options = new QROptions;

		$options->imageBase64 = false;

		$img= $qrcode->render( 'test'  ) ;
 `
		
		creates

`data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAJEAAACRCAIAAABMus10AAAABnRSTlMA/wD/AP83WBt9AAAACXBIWXMAAA7EAAAOxAGVKw4bAAACd0lEQVR4nO3dQW7jIABA0fFo7n/ldFnJC0sMBvu3762bJtUXMnUwHJ/P5w8pf5/+AAzTrEezHs16NOvRrEezHs16NOvRrEezHs16NOvRrEezHs16NOvRrEeznn8zLz6O467Pce20aOX6fa9XuJxeO/SbbzSzDMc469GsR7MezXqm5iAnNy5vHZplrJs4bPuLhhhnPZr1aNajWc+dc5CToavu0NV+6F7GU/OIdQ8cGWc9mvVo1qNZz8I5yFO2TUmeYpz1aNajWY9mPT9hDjKzPKTIOOvRrEezHs16Fs5B1l38n5pWvGQ6Y5z1aNajWY9mPXfOQbY9VDKzHmToq5ltf9EQ46xHsx7NejTrOV7yv/0M38Xwdpr1aNajWc9j+4PcuIvHzKO6626LrFsaa5z1aNajWY9mPfv2BzldhGd2AJm5gM9MOp7a0OzEOOvRrEezHs169u0PMnQB3/ZA7Utumgwxzno069GsR7Oefc/mzkwrtq34uHGrtJnffM0469GsR7MezXqm5iAzx7hse3pl3eKRpxhnPZr1aNajWc9j58XM3DXYtle79SDcQ7MezXo060mem7ttecg717AYZz2a9WjWo1lP8tzc6x9ed7fCfRD+k2Y9mvVo1pM8N3fmfbftALJuHa1x1qNZj2Y9mvUkz6y78fJ+4+LXbef1Gmc9mvVo1qNZT3IOcjLzeM6697UehG+a9WjWo1lP8tzck3WLR2Yez5n54WvGWY9mPZr1aNaTPDf32szK0XW3NnwX86tp1qNZj2Y9P+Hc3N/GOOvRrEezHs16NOvRrEezHs16NOvRrEezHs16NOvRrEezHs16NOvRrEezni+wifc/oOyB/wAAAABJRU5ErkJggg==`

**Comment by codemasher** [comment]
Hey, thanks for the report! The [authenticator example](https://github.com/chillerlan/php-qrcode/blob/main/examples/authenticator.php) does exactly what you're trying to do, so  can you tell me more details about what version you are using and maybe post a full example that allows me to reproduce the issue?

**Comment by alanef** [comment]
Hi tahnks for a very quick response.

I'm using 4.3.4  
I see that Authenticator  is in 5.0-Beta

I'll try a few things and see what happens. ( I only had GD in my Docker so just installed IMGMAGIK but that made no difference )

**Comment by alanef** [comment]
OK doing this ( which is what the docs indicate )

  doesn't work  ( and I was wondering how it could )
  
  `$qrcode =new QRCode;

	$options = new QROptions;

	$options->imageBase64 = false;

	$img= $qrcode->render( 'test'  ) ;
`

But doing

`$options = new QROptions(['imageBase64'         => false]);
		$qrcode =new QRCode($options);
		`
does work  as in date - base 64 has gone - but it is generating PDF  rather than SVG

**Reply by codemasher** [reply]
Wait a second, i think there's some things gotten mixed up there. First off - the default output has changed between v4.x (PNG) and v5.x (SVG).
So if you don't explicitly specify the `outputType` option, the default output will be used, which means if you turn off base64 in v4.x, you'll get the raw PNG.

Try the following:
```php
$options = new QROptions;
$options->outputType= QROutputInterface::MARKUP_SVG; // v4.x: QRCode::OUTPUT_MARKUP_SVG
$options->imageBase64 = false;

// ...
```

**Comment by alanef** [comment]
OK worked in 5 beta but you explained why!  Thanks.  How close is 5.0 beta  - just wondering if I shoudl stay on teh beta as not production yet

**Reply by codemasher** [reply]
I don't have an estimate yet, in fact, it was supposed to be released already but my brain gets in the way every now and then... Currently I'm rewriting the documentation, which prompts some code changes here and there, but I'm positive that it'll be ready for release in the next weeks.
